scripts/lint_prowjobs/main.go

```diff
@@ -145,23 +173,37 @@ func getFilesChanged(gitRoot string, pullBaseSha string, pullPullSha string) ([]
 	return presubmitFiles, err
 }
 
-func unmarshalJobFile(filePath string, jobConfig *config.JobConfig) (*UnmarshaledJobConfig, error, error) {
+func unmarshalJobFile(filePath string, jobConfig *config.JobConfig) *UnmarshaledJobConfig {
 	if _, err := os.Stat(filePath); os.IsNotExist(err) {
-		return nil, nil, nil
+		return nil
 	}
+
 	unmarshaledJobConfig := new(UnmarshaledJobConfig)
 	unmarshaledJobConfig.GithubRepo = strings.Replace(filepath.Dir(filePath), "jobs/", "", 1)
 	unmarshaledJobConfig.FileName = filepath.Base(filePath)
+	unmarshaledJobConfig.FileContents = unmarshalYamlFile(filePath, &jobConfig)
+	unmarshaledJobConfig.ProwjobConfig = jobConfig
+
+	return unmarshaledJobConfig
+}
 
+func unmarshalYamlFile(filePath string, data interface{}) string {
 	fileContents, fileReadError := ioutil.ReadFile(filePath)
-	unmarshaledJobConfig.FileContents = string(fileContents)
 
-	fileUnmarshalError := yaml.Unmarshal(fileContents, &jobConfig)
-	unmarshaledJobConfig.ProwjobConfig = jobConfig
+	if fileReadError != nil {
+		log.Fatalf("Error reading contents of %s: %v", filePath, fileReadError)
+	}
+
+	unmarshalError := yaml.Unmarshal([]byte(fileContents), data)
```

Comment:
nit: The `ioutil.ReadFile` method returns a byte slice, so is it necessary to cast it to type byte slice?

Reply:
Fixed.